controllers/volumereplicationgroup_controller.go

```diff
@@ -466,15 +466,24 @@ func (v *VRGInstance) restorePVs() error {
 	success := false
 
 	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
-		pvList, err := v.listPVsFromS3Store(s3ProfileName)
+		pvList, err := v.fetchPVClusterDataFromS3Store(s3ProfileName)
 		if err != nil {
-			v.log.Error(err, "failed to retrieve PVs from S3 store", "ProfileName", s3ProfileName)
+			v.log.Error(err, fmt.Sprintf("error fetching PV cluster data from S3 profile %s", s3ProfileName))
 
 			continue
 		}
 
 		v.log.Info(fmt.Sprintf("Found %d PVs", len(pvList)))
 
+		err = v.sanityCheckPVClusterData(pvList)
```

Comment:
We should retain the map generated here and pass it to the next profile we download PVs from? Such that if there is a conflict across PVs in S3Store-1 and S3Store2, we still detect it?

Reply:
We go to the next profile only when a download from an S3 profile fails.  In other words, we don't go to the next profile if a download from an S3 profile succeeds.